### PR TITLE
[PEFT] Implement `QuantedLoRALinear`

### PR DESCRIPTION
**Pull request by sijunhe**

### CODE REVIEW

File: tests/peft/test_quant_lora.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    def test_quant_model_forward(self):
+        lora_config = LoRAConfig(
+            target_modules=[".*q_proj.*", ".*v_proj.*"],
+            r=4,
+            lora_alpha=8,
+        )
+        model = AutoModel.from_pretrained("__internal_testing__/tiny-random-bert")
+        lora_model = LoRAModel(model, lora_config)
+        q_config = QuantConfig(activation=None, weight=None)
+        q_config.add_type_config(LoRALinear, weight=FakeQuanterWithAbsMaxObserver(moving_rate=0.9))
```
Comment (zzjjay) [inline tests/peft/test_quant_lora.py:137]: 在这之前添加 
q_config.add_qat_layer_mapping(LoRALinear, QuantedLoRALinear)

**Review by zzjjay — APPROVED** [review]
LGTM

File: paddlenlp/peft/lora/lora_quant_layers.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from paddle import nn
+from paddle.nn import functional as F
+from paddle.nn.quant.format import ConvertibleQuantedLayer
+
+
+class QuantedLoRALinear(ConvertibleQuantedLayer):
```
Comment (lugimzzz) [inline paddlenlp/peft/lora/lora_quant_layers.py:21]: 应该在这里https://github.com/PaddlePaddle/PaddleNLP/blob/cd59ecd82193c781e4c16b3e11dd516ef44cc412/paddlenlp/peft/lora/lora_model.py#L310 添加QuantedLoRALinear

File: paddlenlp/peft/lora/lora_quant_layers.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    The only difference is that its inputs are all fake quantized.
+
+    Note:
+        In order for proper quantization of this layer, we do (W + AB)x instead of Wx + ABx as in LoRALinear.
+        The quanted logic is quant(W + AB)x
+    """
+
+    def __init__(self, layer: nn.Layer, q_config):
+        super().__init__()
+        if layer.merge_weights:
```
Comment (lugimzzz) [inline paddlenlp/peft/lora/lora_quant_layers.py:33]: 为什么不使用merge_weights

**Pull request merged**